cli/tests/functional/src/test/java/io/helidon/tests/functional/CliMavenTest.java

```diff
@@ -0,0 +1,677 @@
+            this.port = builder.port;
+            this.mavenExecFile = Path.of(mavenHome.toString(), "apache-maven-" + mavenVersion.version, "bin", "mvn").toString();
+            this.groupId = builder.groupId == null ? DEFAULT_GROUPID : builder.groupId;
+            this.artifactId = builder.artifactId == null ? DEFAULT_ARTIFACTID : builder.artifactId;
+            this.build = builder.build == null ? DEFAULT_BUILD : builder.build;
+            this.version = builder.version == null ? DEFAULT_VERSION : builder.version;
+            this.packageName = builder.packageName == null ? DEFAULT_PACKAGE_NAME : builder.packageName;
+            this.workDirectory = builder.workDirectory == null ? workDir : builder.workDirectory;
+            this.isBatch = builder.isBatch;
+            this.isDebug = builder.isDebug;
+            this.isVerbose = builder.isVerbose;
+            this.isDevProcess = builder.isDevProcess;
+            this.m2Home = mavenHome.resolve("apache-maven-" + mavenVersion.version).toString();
+            this.console = new CapturingApplicationConsole();
+        }
+
+        private void parseMavenArgs() {
+            cmdArgs = List.of(
+                    "-U", //update snapshot
+                    "archetype:generate",
+                    "-DinteractiveMode=false",
+                    "-DarchetypeGroupId=io.helidon.archetypes",
+                    "-DarchetypeArtifactId=helidon",
+                    "-DarchetypeVersion=" + ARCHETYPE_VERSION,
+                    "-DgroupId=" + groupId,
+                    "-DartifactId=" + artifactId,
+                    "-Dpackage=" + packageName,
+                    "-Dflavor=" + flavor,
+                    "-Dbase=" + archetype,
+                    "-Dname=" + artifactId);
+        }
+
+        private void parseJavaArgs() {
+            applyMode();
+            cmdArgs.add("--build");
+            cmdArgs.add(build);
+            cmdArgs.add("--version");
+            cmdArgs.add(version);
+            cmdArgs.add("--flavor");
+            cmdArgs.add(flavor);
+            cmdArgs.add("--archetype");
+            cmdArgs.add(archetype);
+            cmdArgs.add("--groupid");
+            cmdArgs.add(groupId);
+            cmdArgs.add("--artifactid");
+            cmdArgs.add(artifactId);
+            cmdArgs.add("--package");
+            cmdArgs.add(packageName);
+            cmdArgs.add("--name");
+            cmdArgs.add(artifactId);
+        }
+
+        private void applyMode() {
+            if (isDebug) {
+                cmdArgs.add("--debug");
+            }
+            if (isVerbose) {
+                cmdArgs.add("--verbose");
+            }
+            if (isBatch) {
+                cmdArgs.add("--batch");
+            }
+        }
+
+        void runInitCommand() {
+            parseJavaArgs();
+            List<String> args = new LinkedList<>(List.of("-jar", jarCliPath, "init"));
+            args.addAll(cmdArgs);
+
+            app = platform.launch(
```

Comment:
We should be able to use io.helidon.build.common.maven.MavenCommand instead of bedrock for this.